### PR TITLE
[gasol] Add new port

### PR DESCRIPTION
**Pull request by mayankkt9**
**Describe the pull request**

- What does your PR fix? Fixes issue #9507 
support for new port GASol - A general Genetic Algorithm Solver in C++

- Which triplets are supported/not supported? Have you updated the CI baseline?
tested x86-windows and osx

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/gasol/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
+vcpkg_from_github(
```
Comment (JackBoosY) [inline ports/gasol/portfile.cmake]: If this port does not support arm/uwp, please add `vcpkg_fail_port_install(ON_TARGET "UWP" ON_ARCH "arm" "arm64")` here.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
x64-windows/x64-windows-static build failure:
```
error MSB4126: The specified solution configuration "Release|x64" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties
```

**Comment by LilyWangL** [comment]
Pinging @mayankkt9 for response. Is work still being done for this PR?

**Comment by mayankkt9** [comment]
> Pinging @mayankkt9 for response. Is work still being done for this PR?

I need help regarding what should I do next

**Review by JackBoosY — APPROVED** [review]

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1678,6 +1681,8 @@ speexdsp:x64-linux=fail
 speexdsp:x64-osx=fail
 spirv-tools:arm-uwp=fail
 spirv-tools:x64-uwp=fail
+# conflicts with nanovg
+stb:x86-windows=ignore
```
Comment (strega-nil) [inline scripts/ci.baseline.txt:1685]: we probably want to skip `nanovg` instead; `stb` is quite important, I think

**Comment by strega-nil** [comment]
Alright, this looks good to me! Thanks @mayankkt9

**Pull request merged**